Dynatrace/dynatrace-operator
# Add CRD to helm chart

# Description

Adds dynatrace-operator CRD to helm chart.
Changes Makefile workflow for yamls generation, as shown on diagram:
![image](https://user-images.githubusercontent.com/35979436/169807987-604be630-7881-4a2c-836f-c15996626919.png)

### Out of scope:
- integration with external services like Rancher, Google Marketplace

## How can this be tested?
1. `make test`
2. `make helm-test`
3. `helm install dynatrace-operator config/helm/chart/default -f ../helm/values-no-csi.yaml --set installCRD=true --atomic --create-namespace -n dynatrace`
4. `helm install dynatrace-operator config/helm/chart/default -f ../helm/values-with-csi.yaml --set installCRD=true --atomic --create-namespace -n dynatrace`
5. 
- change crd
- `make manifests`
- `helm upgrade dynatrace-operator config/helm/chart/default --reuse-values --atomic -ndynatrace`
- check if upgrade succeeded and if crd is changed: `k get crd dynakubes.dynatrace.com -o yaml`


## Checklist
